src/components/TermsDialog.tsx

```diff
@@ -0,0 +1,70 @@
+import { useState } from 'react';
+import {
+  isTermsDialogShowingAtom,
+  latestTermsIndex,
+  localStorageStore,
+  networkConfigAtom,
+  walletServiceAtom,
+} from 'store/app';
+import { useAtom, useAtomValue } from 'jotai';
+import { useStore } from 'zustand';
+import ActionsDialog from './ActionsDialog';
+
+const TermsDialog = () => {
+  const walletService = useAtomValue(walletServiceAtom);
+  const { url } = useAtomValue(networkConfigAtom);
+  const { setLatestTermsAgreedIndex } = useStore(localStorageStore);
+  const [isTermsDialogShowing, setIsTermsDialogShowing] = useAtom(
+    isTermsDialogShowingAtom,
+  );
+  const [isChecked, setIsChecked] = useState(false);
+
+  const proceed = () => {
+    setLatestTermsAgreedIndex(latestTermsIndex);
+    setIsTermsDialogShowing(false);
+    walletService.connect(url);
+  };
+
+  const body = (
+    <div className="mt-2 p-1 max-h-96 overflow-y-auto">
+      <label className="cursor-pointer">
+        <input
+          type="checkbox"
+          className="accent-purple-500 cursor-pointer mr-2"
+          checked={isChecked}
+          onClick={() => setIsChecked(isChecked => !isChecked)}
+        />
+        <span>
+          By clicking here you are indicating that you have read and agree to
+          our{' '}
+          <a
+            className="text-blue-500 hover:text-blue-700 underline transition-colors"
+            target="inter_psm_disclaimer"
+            href="https://docs.inter.trade/disclaimer"
```

Comment:
may change independently of behavior. should be a const. maybe in a config module